core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java

```diff
@@ -1067,20 +1067,22 @@ public String getModuleCallables(String packageName,
             return null;
         }
 
+        GsonBuilder builder = new GsonBuilder();
+        Gson gson = builder.create();
+
         // Go through each callable, parse fasten uri, insert signature.
         for (Object j : json) {
-            JSONObject jObj = (JSONObject) j;
-            var uri = jObj.getString("fasten_uri");
-
             try {
-                var uriObject = FastenUriUtils.parsePartialFastenUri(uri);
-                jObj.put("method_name", uriObject.get(2));
-                jObj.put("method_args", uriObject.get(3));
+                JSONObject jObj = (JSONObject) j;
+                var partialUri = jObj.getString("fasten_uri");
+                var fullUri = FastenUriUtils.generateFullFastenUri("mvn", packageName, packageVersion, partialUri);
+                var uriObj = new FastenJavaURI(fullUri);
+                var js =  gson.toJson(uriObj);
+                jObj.put("fasten_uri", new JSONObject(js));
```

Comment:
Haha I didn't even notice it, but believe me or not, it works as expected. I assume because objects are reference types, so jObj references entities inside json.

But I do agree it's f'ed up in terms of readability and maintainability. I thought of any other way of writing this logic, but everything comes up a bit more messy or with more code. Would a comment explaining that it changes json through referencing be enough? Or you do want me to re-write this code completely?

Reply:
Well, just define a jsonarray called result and add jObj to it in the last line of the for loop and return result instead. The current implementation is very misleading. Also adding comments is not a clean solution, code should be self-explanatory. Instead of adding 3 lines of comments adding these 3 lines of code is better IMO.

Reply:
@ashkboos changed. Please, have a look.